pkg/apis/pipeline/v1beta1/pipelinerun_validation.go

```diff
@@ -101,3 +97,30 @@ func (ps *PipelineRunSpec) Validate(ctx context.Context) (errs *apis.FieldError)
 
 	return errs
 }
+
+func validateSpecStatus(status PipelineRunSpecStatus, cfg *config.Config) *apis.FieldError {
+	alphaAPIFieldsEnabled := cfg.FeatureFlags.EnableAPIFields == config.AlphaAPIFields
+	switch status {
+	case "":
+		return nil
+	case PipelineRunSpecStatusPending,
+		PipelineRunSpecStatusCancelledDeprecated:
+		return nil
+	case PipelineRunSpecStatusCancelled,
+		PipelineRunSpecStatusCancelledRunFinally,
+		PipelineRunSpecStatusStoppedRunFinally:
+		if alphaAPIFieldsEnabled {
+			return nil
+		}
+	}
+	if alphaAPIFieldsEnabled {
```

Comment:
Sorry to come back to this again. Here when the `alphaFields` is not enabled, we might have an error without any information given to the user. Would be nice to have a specific error in that case stating that it fails because the `alphaFields` are not enabled.

There is a helper function you may use:
https://github.com/tektoncd/pipeline/blob/b86a9a2204d988fbd2b187f220d64b5a102c1ca3/pkg/apis/pipeline/v1beta1/version_validation.go#L30